core/trino-main/src/main/java/io/trino/sql/planner/PartitioningHandle.java

```diff
@@ -31,6 +31,10 @@
     private final Optional<CatalogHandle> catalogHandle;
     private final Optional<ConnectorTransactionHandle> transactionHandle;
     private final ConnectorPartitioningHandle connectorHandle;
+    /**
+     * False if rows for a given partition can be distributed between multiple streams.
+     */
+    private final boolean deterministic;
```

Comment:
This is not a great name for this property. What if it has multiple streams but it is deterministic? I think the analogy here is the "arbitrary" partitioning that can happen at the global level. It might not be possible to describe the partitioning in terms of columns, but it's still partitioned on some unknown function.